### PR TITLE
upgrade feature store contract

### PR DESCRIPTION
**Pull request by runhli**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/identity/_credentials/_AzureMLSparkOnBehalfOfCredential.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def _get_client_args(**kwargs) -> Optional[dict]:
 
     token_service_endpoint = os.environ.get("AZUREML_SYNAPSE_TOKEN_SERVICE_ENDPOINT")
     obo_access_token = os.environ.get("AZUREML_OBO_CANARY_TOKEN")
+    obo_endpoint = os.environ.get("AZUREML_OBO_USER_TOKEN_FOR_SPARK_RETRIEVAL_API") or "getuseraccesstokenforspark"
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/identity/_credentials/_AzureMLSparkOnBehalfOfCredential.py:59]: You can use `get(key, value)` for default value if key is not found

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/identity/_credentials/_AzureMLSparkOnBehalfOfCredential.py]: updated

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_store_entity/feature_store_entity.py [hunk 2]
```diff
@@ -33,6 +32,7 @@ def __init__(
         name: str,
         version: str,
         index_columns: List[DataColumn],
+        stage: Optional[str] = None,
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_store_entity/feature_store_entity.py:35]: Can you add doc string for this parameter?

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_store_entity/feature_store_entity.py]: Added for feature set and feature store entity.

**Review by luigiw — APPROVED** [review]

**Pull request merged**